### PR TITLE
ARROW-3843: [C++][Python] Allow a "degenerate" Parquet file with no columns

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

**Comment by xhochy** [comment]
@wesm Tests are failing.

**Comment by wesm** [comment]
Will fix

**Comment by wesm** [comment]
This should be good now

File: cpp/src/parquet/schema.cc [hunk 1]
```diff
@@ -370,9 +370,14 @@ void PrimitiveNode::ToParquet(void* opaque_element) const {
...
   const SchemaElement& root = elements_[0];
 
-  // Validate the root node
   if (root.num_children == 0) {
-    throw ParquetException("Root node did not have children");
+    if (length_ == 1) {
+      // Degenerate case of Parquet file with no columns
+      return GroupNode::FromParquet(static_cast<const void*>(&root), next_id(), {});
+    } else {
+      throw ParquetException(
```
Comment (fsaintjacques) [inline cpp/src/parquet/schema.cc]: Didn't know we could throw in parquet.

**Review by fsaintjacques — CHANGES_REQUESTED** [review]

**Review by fsaintjacques — APPROVED** [review]

**Comment by fsaintjacques** [comment]
OSX failing tests is due to LLVM7 brew error. AppVeyoer is due to anaconda http mirror failure.

**Pull request closed without merging**